language/move-compiler/src/naming/ast.rs

```diff
@@ -563,6 +563,80 @@ impl fmt::Display for TypeName_ {
     }
 }
 
+impl fmt::Display for Type_ {
+    fn fmt(&self, f: &mut fmt::Formatter) -> std::fmt::Result {
```

Comment:
Type_ intentionally shouldn't have display. There is a lot of contextual information in here that you probably don't want to display, Like the abilities for a type parameter :) 

If you want this, I would rework `core::error_format_impl_`

Reply:
I am on the phone right now but I am trying to get as much info as I can so that I can start working on it at an arbitrary time. What would be your preferred solution then? I can always just turn it into type_to_string function (and perhaps move it to symbolicator itself)?

Reply:
I think you probably want to re-use as much as you can from `core::error_format`? Or you really want/need printing of types that differs from the compiler?

Reply:
I will take a look at `core::error_format` when I am back at my laptop, but since we don't want this to be an implementation of the `Display` trait, I am not sure if it's possible to package it as part of `core::error_format` or in some different way and just reuse parts of `core::error_format`

Reply:
Not sure why you need Display, but if you really do, you can always do something like `MyTypeWrapper(Type)`and `impl Display for MyTypeWrapper { ... /* use core::error_format */ ... }`

The reason `Type` doesn't implement display is to avoid accidents (as you need the substitution map to display non expanded types)

Reply:
> Not sure why you need Display, but if you really do, you can always do something like `MyTypeWrapper(Type)`and `impl Display for MyTypeWrapper { ... /* use core::error_format */ ... }`

I don't really need it, I simply though it would be the cleanest solution considering that some other AST nodes implement `Display` as well. I will simply turn this into `type_to_string`.

Reply:
Do we need another type_to_string though? That is what `error_format` is. The name is simply because it assumes that string will be used in an error

Reply:
OK, now that I had a chance to actually take a look at the code, I understand the whole deal with `error_format` is, and I am not sure if I want to keep the exact format for type-on-hover (I also don't want to re-work `error_format` as this would have a further reaching consequences for how compiler errors are displayed). An example of what I think is unnecessary in the `error_format` output for the type-on-hover application is how it displays `TypeName_::ModuleType` - while I understand that one would find it in some cases useful to have both numeric address and the name (e.g., `(some_module=0xCAFE)::SomeStruct`) it feels a bit too verbose (and not super intuitive for me) for type-on-hover  where `some_module::SomeStruct` feels totally sufficient. Admittedly I was at least a bit inspired by how IntelliJ plugin displays these.

That being said the differences are fairly minor, so there are two solutions, and it would be great to agree on one (I don't have a super strong preference) before I do another edit:
- use `error_format` and live with some potential minor clutter
- have a customized format for type-on-hover (we don't have to add anything to the compiler core for it - it can become a part of displaying `IdentType` in the symbolicator itself)

Reply:
The verbosity is necessary for errors since you can construct a state where it says `a::m::T` is not compatible with `a::m::T`, and it is because the addresses are different, e.g. `(a=0x1)::m::T` vs `(a=0x2)::m::T`. 

Happy to make the error_format a bit more robust/configurable, but it also might make sense for you to do your own thing on the symbols side

Reply:
I went the route of doing this on the symbols side. When thinking about it some more, I figured it would actually make sense to make the IDE display more susceptible to user feedback by keeping it separate from compiler error messages (put a comment along those lines in the code as well).